### DISCUSSION TITLE
redownloading same image across tasks

### DISCUSSION THREAD

**Discussion by jmccann** [concourse/concourse#6191] [Help & Support] [answered]
I just noticed a job that uses the same image multiple times re-download the image in a task right after another task had just downloaded the image.  I've seen this happen a few times.  This seems more common on some images then others.

I assume a job runs on a single worker.  Maybe my assumption is wrong.  If tasks within a job can run across multiple workers that would help explain the above.

But it does seem a certain image gets re-downloaded more often while some other images don't.  Maybe due to the size of the image (several hundred MB).

Is there a way to allow images to stick around longer on workers?  My workers have more then enough free space.  Or something I can do to figure out why I'm constantly having to re-download certain images?  This behavior is causing jobs to take longer to complete.

Thanks!

**Comment by taylorsilva** [comment, marked as answer]
>I assume a job runs on a single worker. Maybe my assumption is wrong. If tasks within a job can run across multiple workers that would help explain the above.

A job is not guaranteed to run on a single worker. Each step in a job can potentially run on another worker.

You could try enabling global resources: https://concourse-ci.org/global-resources.html
I _think_ that may help avoid having multiple tasks pull the same image.

**Reply by jmccann** [reply]
@taylorsilva Thanks for clarifying that a job could run across multiple workers.  I'm guessing that's what's happening then.  👍
